core/templates/dev/head/domain/collection/EditableCollectionBackendApiServiceSpec.js

```diff
@@ -0,0 +1,120 @@
+    UndoRedoService = $injector.get('UndoRedoService');
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    $httpBackend = $injector.get('$httpBackend');
+
+    // Sample collection object returnable from the backend
+    sampleDataResults = {
+      collection: {
+        id: '0',
+        title: 'Collection Under Test',
+        category: 'Test',
+        objective: 'To pass',
+        schema_version: '1',
+        nodes: [{
+          exploration_id: '0',
+          prerequisite_skills: [],
+          acquired_skills: []
+        }],
+        next_exploration_ids: [],
+        completed_exploration_ids: []
+      }
+    };
+  }));
+
+  afterEach(function() {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should successfully fetch an existing collection from the backend',
+      function() {
+    var successHandler = jasmine.createSpy('success');
+    var failHandler = jasmine.createSpy('fail');
+
+    $httpBackend.expect('GET', '/collection_editor_handler/data/0').respond(
+      sampleDataResults);
+    EditableCollectionBackendApiService.fetchCollection('0').then(
+      successHandler, failHandler);
+    $httpBackend.flush();
+
+    expect(successHandler).toHaveBeenCalledWith(sampleDataResults.collection);
+    expect(failHandler).not.toHaveBeenCalled();
+  });
+
+  it('should use the rejection handler if the backend request failed',
+      function() {
+    var successHandler = jasmine.createSpy('success');
+    var failHandler = jasmine.createSpy('fail');
+
+    // Loading a collection the first time should fetch it from the backend.
+    $httpBackend.expect('GET', '/collection_editor_handler/data/1').respond(
+      500, 'Error loading collection 1.');
+    EditableCollectionBackendApiService.fetchCollection('1').then(
+      successHandler, failHandler);
+    $httpBackend.flush();
+
+    expect(successHandler).not.toHaveBeenCalled();
+    expect(failHandler).toHaveBeenCalledWith('Error loading collection 1.');
+  });
+
+  it('should update a collection after fetching it from the backend',
+      function() {
+    var successHandler = jasmine.createSpy('success');
+    var failHandler = jasmine.createSpy('fail');
+
+    // Loading a collection the first time should fetch it from the backend.
+    $httpBackend.expect('GET', '/collection_editor_handler/data/0').respond(
+      sampleDataResults);
+    coll = EditableCollectionBackendApiService.fetchCollection('0');
+    $httpBackend.flush();
+    originalColl = angular.copy(coll);
+
+    sampleDataResults.collection.title = 'New Title';
+    sampleDataResults.collection.schema_version = '2';
+
+    $httpBackend.expect('PUT', '/collection_editor_handler/data/0').respond(
+      sampleDataResults);
+
+    // Send a request to update collection
+    EditableCollectionBackendApiService.updateCollection(
+      '0', '2', 'New Title', UndoRedoService.getChangeList()).then(
```

Comment:
Firstly, I don't think you need ``originalCol``. Also, consider cloning the collection returned and changing that, rather than changing ``sampleDataResults``. For instance:

```javascript
var collection = EditableCollectionBackendApiService.fetchCollection('0');
$httpBackend.flush();

collection.title = 'New Title';

$httpBackend.expect('PUT', '/collection_editor_handler/data/0').respond(
  collection);

// Send a request to update the collection
EditableCollectionBackendApiService.updateCollection(
  collection.id, collection.version, collection.title, []).then(
    successHandler, failHandler);
$httpBackend.flush();

...
```